geoportailv3/static/js/locationControl.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * @fileoverview This file defines the geolocation control.
+ *
+ */
+
+goog.require('ol.Feature');
+goog.require('ol.FeatureOverlay');
+goog.require('ol.Geolocation');
+goog.require('ol.control.Control');
+
+
+goog.provide('app.LocationControl');
+
+
+
+/**
+ * @constructor
+ * @extends {ol.control.Control}
+ * @param {Object=} opt_options Control options.
+ */
+app.LocationControl = function(opt_options) {
+
+  var options = opt_options || {};
+  var className = goog.isDef(options.className) ? options.className :
+      'location-button';
+  /**
+   * @type {?ol.Geolocation}
+   * @private
+   */
+  this.geolocation_ = null;
+  /**
+   * @type {?ol.FeatureOverlay}
+   * @private
+   */
+  this.featureOverlay_ = null;
+  this.positionPoint_ = new ol.geom.Point([0, 0]);
+
+  var label = goog.isDef(options.label) ? options.label : 'L';
+  var tipLabel = goog.isDef(options.tipLabel) ?
+      options.tipLabel : 'Location';
+  var button = goog.dom.createDom(goog.dom.TagName.BUTTON, {
+    'type': 'button',
+    'title': tipLabel
+  }, label);
+
+
+
+  var cssClasses = className + ' ' + ol.css.CLASS_UNSELECTABLE + ' ' +
+      ol.css.CLASS_CONTROL + ' ' + 'tracker-off';
+
+  /**
+   * @type {!Element}
+   */
+  this.element = goog.dom.createDom(goog.dom.TagName.DIV, cssClasses, button);
+
+  goog.events.listen(button, goog.events.EventType.CLICK,
+      this.handleClick_, false, this);
+
+  goog.events.listen(button, [
+    goog.events.EventType.MOUSEOUT,
+    goog.events.EventType.FOCUSOUT
+  ], function() {
+    this.blur();
+  }, false);
+
+  goog.base(this, {
+    element: this.element,
+    target: options.target
+  });
+
+
+};
+goog.inherits(app.LocationControl, ol.control.Control);
+
+
+/**
+ * @param {goog.events.BrowserEvent} event The event to handle
+ * @private
+ */
+app.LocationControl.prototype.handleClick_ = function(event) {
+  event.preventDefault();
+  this.handleCenterToLocation_();
+};
+
+
+/**
+ * @private
+ */
+app.LocationControl.prototype.handleCenterToLocation_ = function() {
+  if (this.geolocation_ == null) {
+    this.initGeoLocation_();
+  }
+
+  if (!this.geolocation_.getTracking()) {
+    this.initFeatureOverlay_();
+    this.getMap().getView().setZoom(17);
+    this.geolocation_.setTracking(true);
+  }else {
+    this.clearFeatureOverlay_();
+    this.geolocation_.setTracking(false);
+
+  }
+};
+
+
+/**
+ *
+ * @private
+ */
+app.LocationControl.prototype.initGeoLocation_ = function() {
+
+  var map = this.getMap();
+  var view = map.getView();
+  this.geolocation_ = /** @type {ol.Geolocation} */ (new ol.Geolocation({
+    projection: view.getProjection()
+  }));
+
+  this.geolocation_.on('change:tracking', goog.bind(function(e) {
+
+    if (this.geolocation_.getTracking()) {
```

Comment:
Again, I'd recommend to be more explicit by using `goog.isDef` or `goog.isNull` or `goog.isDefAndNotNull`.